.github/workflows/code-quality.yaml

```diff
@@ -83,8 +89,23 @@ jobs:
                     php-version: 8.1
                     tools: flex
 
+            -   name: Get composer cache directory
+                id: composer-cache
+                run: |
+                    echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+            - name: Cache packages dependencies
+              uses: actions/cache@v4
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: ${{ runner.os }}-composer-packages-${{ matrix.php-version }}-${{ matrix.dependency-version }}-${{ matrix.symfony-version }}-${{ matrix.minimum-stability }}-${{ hashFiles('src/**/composer.json') }}
+                  restore-keys: |
+                      ${{ runner.os }}-composer-packages-${{ matrix.php-version }}-${{ matrix.dependency-version }}-${{ matrix.symfony-version }}-${{ matrix.minimum-stability }}
+
             - name: Install root dependencies
-              run: composer install
+              uses: ramsey/composer-install@v3
+              with:
+                  working-directory: ${{ github.workspace }}
```

Comment:
Manually caching Composer dependencies is still necessary because dependencies are parallelly installed through `composer`, which AFAIK is not possible with `ramsey/composer-install`